File: Algorithm/数组/一维数组/反转数组/反转数组.ts
// 反转数组

function reverse(numbers: number[]): number[] {
  let left: number = 0;
  let right: number = numbers.length - 1;
  while (left <= right) {
    const temp = numbers[left];
    numbers[left] = numbers[right];
    numbers[right] = temp;
    left++;
    right--;
  }
  return numbers;
}

// test
const res = reverse([1, 2, 4, 3]);
console.log(res);

File: Data Structure & Algorithm/Archive/Algorithm/Two Points/zh/左右指针/反转数组/反转数组.py
# 反转数组

from typing import List


def reverse(numbers: List[int]) -> List[int]:
    left = 0
    right = len(numbers) - 1
    while left < right:
        temp = numbers[left]
        numbers[left] = numbers[right]
        numbers[right] = temp
        left = left + 1
        right = right - 1
    return numbers


# test
nums = [-1, 0, 3, 5, 9, 12]
res = reverse(nums)
print(res)